crowbar_framework/app/models/service_object.rb

```diff
@@ -669,6 +669,10 @@ def select_nodes_for_role(all_nodes, role, preferred_intended_role = nil)
       valid_nodes = preferred_all_nodes unless preferred_all_nodes.empty?
     end
 
+    if role_constraints[role] && role_constraints[role].has_key?("count")
+      valid_nodes = valid_nodes.take(role_constraints[role]["count"])
```

Comment:
according to http://www.ruby-doc.org/core-2.2.0/Array.html#method-i-take take does an exception on negative values. please check for >=0 first.